src/pages/search/index.tsx

```diff
@@ -114,24 +133,54 @@ export default function Search() {
     });
   };
 
+  const switchShowMode = () => {
+    let nextMode;
+    // If show is undefined we know we are in list view, so nextMode should be map
+    if (!show) {
+      nextMode = ShowMode.MAP;
+    } else {
+      nextMode = show === ShowMode.LIST ? ShowMode.MAP : ShowMode.LIST;
+    }
+
+    const params = getURLSearchParamsFromAsPath(router.asPath);
+    params.set("show", nextMode);
+
+    router.replace(
+      { pathname: router.pathname, query: params.toString() },
+      undefined,
+      {
+        shallow: true,
+      }
+    );
```

Comment:
Personally I would have set up the map view in a different route, e.g. `search/map` for better separation of concerns. Although this approach is nice now in that data can be efficiently shared, the map will likely get its own query in the future, right?

Reply:
Actually moving map to it's own route would make more sense. Even that we were talking about changing the search for map view it totally skipped my mind while implementing this.